Lower right info box improvements

stavrosfa opened the pull request:
1. Added unit thumbnail in the lower right info box
2. Now it also displays the unit's rank, and also if the unit is in a city, the name of the city instead of the underlying terrain info
3. Clicking on it will center on the currently active unit
4. Clicking on it at the end of the turn will end the turn

To be honest this one felt a bit "hacky",  for lack of a better word. If you have any suggestions as to how to improve it, let me know.

stavrosfa committed:
1. Added unit thumbnail in the lower right info box
2. Clicking will center on the currently active unit
3. Clicking at the end of the turn will end the turn

stavrosfa committed:
Whitespace formatting

TomWerner commented:
Very cool! Sorry for the slow review, missed the email. 

My one comment is that I'd rather not duplicate the "get the first frame of the animation" logic that we already have in the city screen: 
https://github.com/C7-Game/Prototype/blob/370b55db9c9f9789975ffcd13f1934876fcc3230/C7/UIElements/CityScreen/CityScreen.cs#L597-L627

Could you share that code for the lower right hand box?

stavrosfa committed:
CityScreen calls AnimationManager to fetch thumbnail textures to display

stavrosfa commented:
No worries, I though we were in some kind of freeze before the release. 

Regarding what you said, you are absolutely right, but I think instead of using the city screen logic in the lower right hand box,  the city screen method should call the method in the AnimantionManager.
Maybe that's what you meant all along, because calling the cityScreen from the LowerRightInfoBox doesn't make sense to me.

TomWerner commented on C7/Animations/AnimationManager.cs:
```diff
@@ -40,6 +40,13 @@ public static string AnimationKey(UnitPrototype unit, MapUnit.AnimatedAction act
 		return AnimationKey(BaseAnimationKey(unit, action), direction);
 	}
 
+	public static readonly Dictionary<string, ImageTexture> AnimationThumbnails = new();
```
Can we ensure these caches are cleared when we clear the other caches? Otherwise we might carry things over between scenarios or when switching the textures being used.

TomWerner commented:
> Maybe that's what you meant all along, because calling the cityScreen from the LowerRightInfoBox doesn't make sense to me.

Yeah, I should have been clearer sorry. I just wanted to make sure the code was in some common location. I agree it wouldn't make sense for the lower right box to depend on the city screen.

stavrosfa committed:
Clear AnimationManager cache before game

stavrosfa committed:
1. Added event for when a (human) unit has moved, to update its position in the LRIB, instead of checking every frame.
2. This event also updates the unit action buttons in the UI
3. Minor reordering in the AnimateAsync method to make sure the event gets called even when animations are disabled

stavrosfa committed:
Whitespace formatting

TomWerner commented:
Still looks good, thanks for the improvements!
